src/main/scala/mesosphere/marathon/core/launcher/impl/TaskLabels.scala

```diff
@@ -1,24 +1,18 @@
 package mesosphere.marathon.core.launcher.impl
 
 import mesosphere.marathon.core.task.Task
+import org.apache.mesos.Protos.Label
 import org.apache.mesos.{ Protos => MesosProtos }
 
+import scala.collection.mutable
+
 object TaskLabels {
   private[this] final val TASK_ID_LABEL = "marathon_task_id"
 
-  def labelsForTask(task: Task): Map[String, String] = labelsForTask(task.taskId)
-  def labelsForTask(taskId: Task.Id): Map[String, String] = Map(TASK_ID_LABEL -> taskId.idString)
-
-  def mesosLabelsForTask(taskId: Task.Id): MesosProtos.Labels = {
-    mesosLabels(labelsForTask(taskId))
+  def taskIdForResource(resource: MesosProtos.Resource): Option[Task.Id] = {
+    ResourceLabels(resource).labels.get(TASK_ID_LABEL).map(Task.Id(_))
```

Comment:
Are we sure that this framework made the reservation?
If people start more than 1 marathon on the same role, we would fight each other.